h/services/group.py

```diff
@@ -73,6 +73,32 @@ def create(self, name, authority, userid, description=None, type_='private'):
 
         return group
 
+    def create_open_group(self, name, authority, userid, hostnames, description=None):
```

Comment:
Where does the responsibility lie for validating that `hostnames` are valid actual hostnames? i.e. sanitation/validation

Reply:
I think that would live in a separate validation layer, like how for annotations we have validation schemas in a separate module: <https://github.com/hypothesis/h/blob/master/h/schemas/annotation.py>. It's up to the views (API view, HTML page view) to run params through validation and then pass them on to the service. See for example [the view for creating annotations](https://github.com/hypothesis/h/blob/master/h/views/api.py#L123-L134).

For creating groups, validation might not be complex enough to merit a separate schema class yet. Maybe just the view (admin page or CLI command) would validation the hostnames inline.

Reply:
(Based on a quick look it looks like currently the only validation that occurs for creating private groups is actually in the groups _model_ code, so that's an option too.)

(P.P.S. Possibly we don't really _need_ to validate them?)